mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp

```diff
@@ -913,6 +931,14 @@ struct WarpOpDeadResult : public OpRewritePattern<WarpExecuteOnLane0Op> {
     // Move the body of the old warpOp to a new warpOp.
     WarpExecuteOnLane0Op newWarpOp = moveRegionToNewWarpOpAndReplaceReturns(
         rewriter, warpOp, newYieldValues, newResultTypes);
+
+    // Simplify the new warp op after dropping dead results.
+    auto simplifyFn = [&](Operation *op) {
```

Comment:
This functor is simple enough; we should just inline it in the `walk` call. (For functors that are complicated it's useful to extract out and give a name to make formatting and logic cleaner.)